Scripts/Get-UplinkDetails.ps1

```diff
@@ -0,0 +1,147 @@
+function Get-UplinkDetails {
+<#	
+    .NOTES
+    ===========================================================================
+    Created by: Markus Kraus
+    ===========================================================================
+    Changelog:  
+    2017.03 ver 1.0 Base Release  
+    2020.03 ver 1.1 Add LLDP Support
+    ===========================================================================
+    External Code Sources: 
+    Get-CDP Version from @LucD22
+    https://communities.vmware.com/thread/319553
+	
+	LLDP PowerCLI Tweak
+	https://tech.zsoldier.com/2018/05/vmware-get-cdplldp-info-from.html
+    ===========================================================================
+    Tested Against Environment:
+    vSphere Version: vSphere 6.7 U3
+    PowerCLI Version: PowerCLI 11.5
+    PowerShell Version: 5.1
+    OS Version: Server 2016
+    Keyword: ESXi, Network, CDP, LLDP, VDS, vSwitch, VMNIC 
+    ===========================================================================
+
+    .DESCRIPTION
+    This Function collects detailed informations about your ESXi Host connections to pSwitch and VDS / vSwitch. 
+    LLDP Informations might only be available when uplinks are connected to a VDS.
+
+    .Example
+    Get-VMHost -Name MyHost | Get-UplinkDetails -Type LLDP | Where-Object {$_.VDS -ne "-No Backing-"}  | Format-Table -AutoSize
+
+    .Example
+    Get-VMHost -Name MyHost | Get-UplinkDetails -Type CDP | Where-Object {$_.VDS -ne "-No Backing-"}  | Format-Table -AutoSize
+
+    .Example
+    Get-Cluster -Name MyCluster | Get-VMHost | Get-UplinkDetails -Type LLDP | Format-Table -AutoSize
+
+    .Example
+    Get-Cluster -Name MyCluster | Get-VMHost | Get-UplinkDetails -Type CDP | Format-Table -AutoSize
+
+    .PARAMETER myHosts
+    Hosts to process
+
+
+#Requires PS -Version 5.0
+#Requires -Modules VMware.VimAutomation.Core, @{ModuleName="VMware.VimAutomation.Core";ModuleVersion="6.3.0.0"}
+#>
+
+[CmdletBinding()]
+param( 
+    [Parameter(Mandatory=$True, ValueFromPipeline=$True, Position=0, HelpMessage = "Hosts to process")]
+        [ValidateNotNullorEmpty()]
+        [VMware.VimAutomation.ViCore.Impl.V1.Inventory.InventoryItemImpl[]] $myHosts,
+    [Parameter(Mandatory=$True, ValueFromPipeline=$False, Position=1, HelpMessage = "Type of infos to collect (CDP / LLDP)")]
+        [ValidateSet("CDP","LLDP")]
+        [String] $Type
+        
+)
+
+Begin {
+
+
+    function Get-Info ($VMhost){
+        $VMhostProxySwitch = $VMhost.NetworkInfo.ExtensionData.ProxySwitch 
+        $VMhostSwitch = $VMhost.NetworkInfo.VirtualSwitch
+
+        $objReport = @()
+        $VMhost| %{Get-View $_.ID} | 
+        %{ Get-View $_.ConfigManager.NetworkSystem} | 
+        %{ foreach($physnic in $_.NetworkInfo.Pnic){ 
```

Comment:
That is quite complicated to me.
`| Foreach-Object {foreeach()}`
My recommendation is to grab the result of network infos in a variable first and then do the iteration on a separate foreach call.